src/state.rs

```diff
@@ -146,6 +157,18 @@ impl<BackendData: Backend + 'static> MagmaState<BackendData> {
             )
             .expect("Failed to init wayland server source");
 
+        loop_handle
+            .insert_source(
+                Timer::from_duration(Duration::from_millis(xcursor.get_curr_image().delay.into())),
```

Comment:
this will only tick for the delay of the initial cursor image, if the cursor changes the timer duration wont change, which is a problem